pkg/dockercreds/access_checker.go

```diff
@@ -14,42 +13,12 @@ import (
 )
 
 func VerifyWriteAccess(keychain authn.Keychain, tag string) error {
-	var auth authn.Authenticator
 	ref, err := name.ParseReference(tag, name.WeakValidation)
 	if err != nil {
 		return errors.Wrapf(err, "Error parsing reference %q", tag)
 	}
 
-	auth, err = keychain.Resolve(ref.Context().Registry)
-	if err != nil {
-		return errors.Wrap(err, "Error resolving credentials")
-	}
-
-	scopes := []string{ref.Scope(transport.PushScope)}
-	tr, err := transport.New(ref.Context().Registry, auth, http.DefaultTransport, scopes)
-	if err != nil {
-		return diagnoseIfTransportError(err)
-	}
-	client := &http.Client{Transport: tr}
-
-	u := url.URL{
-		Scheme: ref.Context().Registry.Scheme(),
-		Host:   ref.Context().RegistryStr(),
-		Path:   fmt.Sprintf("/v2/%s/blobs/uploads/", ref.Context().RepositoryStr()),
-	}
-
-	// Make the request to initiate the blob upload.
-	resp, err := client.Post(u.String(), "application/json", nil)
-	if err != nil {
-		return diagnoseIfTransportError(err)
-	}
-	defer resp.Body.Close()
-
-	if err = transport.CheckError(resp, http.StatusCreated, http.StatusAccepted); err != nil {
-		return err
-	}
-
-	return nil
+	return remote.CheckPushPermission(ref, keychain, http.DefaultTransport)
 }
 
 func VerifyReadAccess(keychain authn.Keychain, tag string) error {
```

Comment:
I'm not sure if we need the `func diagnoseIfTransportError(err error) error` still. This appears to be handling an edge case for artifactory. I think we should remove this or add the artifactory test back. What's the context for this? @matthewmcnew

Reply:
I saw that we're using it in [VerifyReadAccess](https://github.com/pivotal/kpack/blob/44257a70c2c9e9ac5703eadc871c7c9be22cfadc/pkg/dockercreds/access_checker.go#L62).

Reply:
By the way, I don't know if it matters or not but [imgutil is using the ggcr Head function](https://github.com/buildpacks/imgutil/blob/56137f75e24d5b567db1dbb47ab05e958e257ec2/remote/remote.go#L476) in order to check read access while [kpack is using the ggcr Get function](https://github.com/pivotal/kpack/blob/44257a70c2c9e9ac5703eadc871c7c9be22cfadc/pkg/dockercreds/access_checker.go#L61).

Reply:
`imgutil` is probably doing it correctly

Reply:
using get is probably fine

Reply:
let's just add `diagnoseIfTransportError` to the write check. using get should still be fine.

Reply:
Please see a67ce1afb9eda1ec9c5517379b6600b70381372b (I'll merge all of the commits before merging to `main`).